web/frontend/src/components/slam/index.svelte

```diff
@@ -93,6 +94,38 @@ const startDurationTimer = (start: number) => {
   }, 400);
 };
 
+const setMappingMode = async () => {
+  console.log("in setMappingMode, slamModel: ", overrides?.slamModel)
+  console.log("in setMappingMode, overrides.mode: ", overrides?.mappingDetails.mode)
+  console.log("in setMappingMode, mappingMode: ", mappingMode)
+  // modes: 'localize' | 'create' | 'update' | 'undefined'
+  mappingMode = overrides?.mappingDetails.mode || 'undefined';
+  if (overrides?.slamModel !== 'viam:slam:cartographer') {
+    try {
+        const props = await slamClient.getProperties();
+        switch (props.mappingMode) {
+          case slamApi.MappingMode.MAPPING_MODE_CREATE_NEW_MAP:
```

Comment:
Oh, interesting, since these constants exist, can we just use them instead of assigning new string values?

If the issue is that the values of these constants aren't human readable, then I would recommend creating a map to human-readable values:

```ts
const mappingModeToDisplayText = {
  [slamApi.MappingMode.MAPPING_MODE_CREATE_NEW_MAP]: 'create',
  [slamApi.MappingMode.MAPPING_MODE_LOCALIZE_ONLY]: 'localize',
  ...
}
```

```html
<span>{mappingModeToDisplayText[mappingMode]}</span>
```